### PR TITLE
Add isolated flush timer to CommitLogMetrics and ensure writes correspond to single WaitingOnCommit data points

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe for CASSANDRA-16701

### CODE REVIEW

File: src/java/org/apache/cassandra/metrics/CommitLogMetrics.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import org.apache.cassandra.db.commitlog.AbstractCommitLogService;
 import org.apache.cassandra.db.commitlog.AbstractCommitLogSegmentManager;
 
+import java.util.concurrent.TimeUnit;
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/metrics/CommitLogMetrics.java]: unused import

File: src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java [hunk 2]
```diff
@@ -499,13 +503,12 @@ void waitForFinalSync()
...
     }
 
-    void waitForSync(int position, Timer waitingOnCommit)
+    void waitForSync(int position)
     {
         while (lastSyncedOffset < position)
         {
-            WaitQueue.Signal signal = waitingOnCommit != null ?
-                                      syncComplete.register(waitingOnCommit.time()) :
-                                      syncComplete.register();
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java:508]: A general question... why does `AbstractCommitLogService#awaitSyncAt` (called by `PeriodicCommitLogService`) still register the timer context with the syncComplete?

Comment (maedhroz) [reply src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java:508]: `PeriodicCommitLogService` doesn't block writes under normal conditions, so I left that behavior alone. The big motivating factor for this change was making sure that when "batch" or "group" CL are in use, the metric data points correspond to single writes (and vise-versa). It probably wouldn't be harmful to do it for "periodic" as well, but it also doesn't buy much.

**Review by yifan-c — APPROVED** [review]

**Comment by maedhroz** [comment]
Committed in https://github.com/apache/cassandra/commit/5f23e6d766a18782db82d955bf380239990d2c84

**Pull request closed without merging**